src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -44,6 +44,18 @@ function VisProvider(props: Props): ReactElement {
       );
       // The dimensions untouched by the slicing are only the ones of 'x' and 'y'
       const slicedDims = rawDims.filter((d, i) => !isNumber(mapperState[i]));
+      if (
+        mapperState.includes('y') &&
+        mapperState.indexOf('x') < mapperState.indexOf('y')
+      ) {
+        return ndarray(
+          unpack(rawDataArray.pick(...slicingIndices).transpose(1, 0)).flat(
+            Infinity
+          ),
+          slicedDims.reverse()
+        );
+      }
```

Comment:
We expect the first dimension of 2D arrays to be `y`. If not, the array is transposed.